app/controllers/courses_controller.rb

```diff
@@ -16,6 +16,8 @@ class CoursesController < ApplicationController
   before_action :require_permissions, only: %i[notify_untrained
                                                delete_all_weeks]
 
+  after_action :ensure_home_wiki_in_courses_wikis, only: %i[create update]
```

Comment:
Using `after_action` here may be part of the problem you're having, as I think that happens after the view gets rendered, so the JSON representation of the course that gets passed back to the client won't reflect any changes that come from that callback. I suggest calling it explicitly from `update_courses_wikis` instead of using a callback.

Reply:
We don't use the changes from ater_action but yes, I will include it in update_courses_wikis.